docs/en/sql-reference/functions/type-conversion-functions.md

```diff
@@ -3542,173 +3542,312 @@ SELECT toDateTime64('2019-01-01 00:00:00', 3, 'Asia/Istanbul') AS value, toTypeN
 
-- `expr` — [Expression](../syntax.md/#syntax-expressions), returns a value in the [String](../data-types/string.md) data type. ClickHouse expects the textual representation of the decimal number. For example, `'1.111'`.
-- `S` — Scale, the number of decimal places in the resulting value.
+:::note
+Integer overflow can occur if the value of `expr` exceeds the bounds of `Decimal32`: `( -1 * 10^(9 - S), 1 * 10^(9 - S) )`. 
+Excessive digits in a fraction are discarded (not rounded). 
+Excessive digits in the integer part will lead to an exception.
+:::
 
 **Returned value**
 
-A value in the `Nullable(Decimal(P,S))` data type. The value contains:
+- Value of type `Decimal(9, S)`. [Decimal32(S)](../data-types/int-uint.md).
 
-- Number with `S` decimal places, if ClickHouse interprets the input string as a number.
-- `NULL`, if ClickHouse can’t interpret the input string as a number or if the input number contains more than `S` decimal places.
+**Example**
 
-**Examples**
+Query:
+
+```sql
+SELECT
+    toDecimal32(2, 1) AS a, toTypeName(a) AS type_a,
+    toDecimal32(4.2, 2) AS b, toTypeName(b) AS type_b,
+    toDecimal32('4.2', 3) AS c, toTypeName(c) AS type_c
+FORMAT Vertical;
+```
+
+Result:
+
+```response
+Row 1:
+──────
+a:      2
+type_a: Decimal(9, 1)
+b:      4.2
+type_b: Decimal(9, 2)
+c:      4.2
+type_c: Decimal(9, 3)
+```
+
+**See also**
+
+- [`toDecimal32OrZero`](#todecimal32orzero).
+- [`toDecimal32OrNull`](#todecimal32ornull).
+- [`toDecimal32OrDefault`](#todecimal32ordefault).
+
+## toDecimal32OrZero
+
+Like [`toDecimal32`](#todecimal32), this function converts an input value to a value of type [Decimal(9, S)](../data-types/decimal.md) but returns `0` in case of an error.
+
+**Syntax**
+
+```sql
+toDecimal32OrZero(expr, S)
+```
+
+**Arguments**
+
+- `expr` — A String representation of a number. [String](../data-types/string.md).
+- `S` — Scale parameter from [ 1 : 9 ] specifying how many decimal digits the fractional part of a number can have. [UInt8](../data-types/int-uint.md).
+
+Supported arguments:
+- String representations of type (U)Int8/16/32/64/128/256.
+- String representations of type Float32/64.
+
+Unsupported arguments:
+- String representations of Float32/64 values `NaN` and `Inf` (case-insensitive).
+- String representations of binary and hexadecimal values, e.g. `SELECT toDecimal32OrZero('0xc0fe', 1);`.
+
+:::note
+Integer overflow can occur if the value of `expr` exceeds the bounds of `Decimal32`: `( -1 * 10^(9 - S), 1 * 10^(9 - S) )`.
+Excessive digits in a fraction are discarded (not rounded).
+Excessive digits in the integer part will lead to an exception.
+:::
+
+**Returned value**
+
+- Value of type `Decimal(9, S)` if successful, otherwise `0` with `S` decimal places. [Decimal32(S)](../data-types/decimal.md).
+
+**Example**
 
 Query:
 
 ``` sql
-SELECT toDecimal32OrNull(toString(-1.111), 5) AS val, toTypeName(val);
+SELECT
```

Comment:
To keep things simple, could we have just one example query like in `toDecimal32`?

Also, both examples below don't trigger the error case. One example input actually should.